webgpu.h

```diff
@@ -144,11 +144,27 @@ typedef enum WGPUCullMode {
     WGPUCullMode_Force32 = 0x7FFFFFFF
 } WGPUCullMode;
 
+typedef enum WGPUErrorFilter {
+    WGPUErrorFilter_None = 0x00000000,
+    WGPUErrorFilter_Validation = 0x00000001,
+    WGPUErrorFilter_OutOfMemory = 0x00000002,
+    WGPUErrorFilter_Force32 = 0x7FFFFFFF
+} WGPUErrorFilter;
+
+typedef enum WGPUErrorType {
+    WGPUErrorType_NoError = 0x00000000,
```

Comment:
Do we need a "NoError" variant at all for an error type? That doesn't sound right.

Reply:
Yeah, I think it's necessary. Maybe we should rename this enum? It's used when the callback is called but an error has not occurred. In C it makes sense to guarantee all once-callbacks will be called exactly once (even if there's no error), in case the caller allocates memory that it wants to free after the callback runs.

Reply:
ok, something like `WGPUErrorType_None = 0` is fine then

Reply:
Just remembered, it's also needed for popErrorScope if there was no error inside the scope (where in JS we resolve to null).